### PR TITLE
Add function btCollisionObject::setIgnoreCollisionCheck() from bullet 2.83

### PR DESCRIPTION
**Pull request by FlorentMasson**
Add function btCollisionObject::setIgnoreCollisionCheck() from bullet 2.83: https://github.com/bulletphysics/bullet3/commit/7151865c16ba996996206e1fd7869cbb1e7edd8d

Fix a collision check bug with setIgnoreCollisionCheck and CCD: https://github.com/bulletphysics/bullet3/pull/1662/commits/cfa3a6d2aa20209ca88dd5f4a2703237f6ffdecc

### CODE REVIEW

**Comment by kripken** [comment]
Thanks @FlorentMasson , but I think it might be better to upgrade to the new version completely, instead of adding specific functions one by one, which can end up more work (and even more work when eventually doing the full upgrade).

**Comment by FlorentMasson** [comment]
I tried that first, but as the build system looked different in bullet 2.88 or even 2.84, I did not succeed (not mentioning all custom additions marked as XXX)
So... not from me without some guidance...

**Comment by kripken** [comment]
Yeah, if the API changed significantly then this would take some work. Hopefully someone has time to do that.

Meanwhile if merging this is urgent for you, I'm not opposed, but it might be a good idea to add some documentation of this change, like in a changelog or in the files themselves. Which is better depends on the process for moving to a new version, which I guess we don't have a process for atm... so maybe a changelog is enough.

File: buildAll.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+python make.py && mv builds/ammo.js builds/ammo.dev.js
```
Comment (kripken) [inline buildAll.sh:3]: what are the new "dev" builds for?

File: bullet/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp [hunk 2]
```diff
@@ -189,7 +189,7 @@ bool	btCollisionDispatcher::needsCollision(const btCollisionObject* body0,const
 
 	if ((!body0->isActive()) && (!body1->isActive()))
 		needsCollision = false;
-	else if (!body0->checkCollideWith(body1))
+	else if ((!body0->checkCollideWith(body1)) || (!body1->checkCollideWith(body0)))
```
Comment (kripken) [inline bullet/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp:192]: for changes inside bullet code, please add a comment `// XXX EMSCRIPTEN` with an explanation of what changed.

or for a line like this, even
```
#ifdef __EMSCRIPTEN__
..changed line..
#else
..original line..
#endif
```

These will make it much easier to upgrade in the future.

Comment (FlorentMasson) [reply bullet/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp:192]: that's from https://github.com/bulletphysics/bullet3/commit/7151865c16ba996996206e1fd7869cbb1e7edd8d

**Comment by FlorentMasson** [comment]
Sorry, I was not aware my subsequent commits would be added to the PR. I used the repo for further custom needs.
I'm closing it as it looks the PR will never be merged anyway...

File: bullet/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp (hunk 2 above)
Comment (FlorentMasson) [reply bullet/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp:192]: that's from https://github.com/bulletphysics/bullet3/commit/7151865c16ba996996206e1fd7869cbb1e7edd8d

**Comment by kripken** [comment]
I would be ok with a PR that backports patches from newer bullet, sorry, in my last response I didn't realize that that is what that specific line was.

As discussed earlier we should document what backporting was done.

**Pull request closed without merging**